tests/executables_src/testIntcHandler.cpp

```diff
@@ -0,0 +1,376 @@
+/* probably this is the first one to be implemented
+ * if neither (SIE and CIE) nor MIR are present, or only IER is there, it writes 1<<N to IER and clears with 1<<N
+ * if neither ICR nor IAR are present: INTC writes the according bit mask to ISR
+ */
+BOOST_AUTO_TEST_CASE(testIERwithISR) { // TEST0
+  ChimeraTK::Device device;
+  device.open(std::string{testCdd});
+
+  BOOST_TEST(device.isOpened() == true);
+  auto accInterrupt =
+      device.getVoidRegisterAccessor("!0:4", {ChimeraTK::AccessMode::wait_for_new_data}); // See DAQ_READY
+  auto dummyInterrupt = device.getVoidRegisterAccessor("DUMMY_INTERRUPT_0"); // for triggering the dummy interrupt
+
+  device.activateAsyncRead();
+
+  BOOST_TEST(readWithTimeout(accInterrupt)); // initial value after activation
+
+  auto ier = device.getScalarRegisterAccessor<int>("TEST0/IER");
+
+  ier.read();
+
+  BOOST_TEST(0x10 == ier);
+
+  auto isr = device.getScalarRegisterAccessor<uint32_t>("TEST0/ISR");
+  // prepare the status before sending the interrupt
+  // set one more bit to be sensitive to the handshake (need to see changes)
+  isr = 0x110;
+  isr.write();
+
+  dummyInterrupt.write();
+
+  // wait until interrupt handler is done
+  BOOST_TEST(readWithTimeout(accInterrupt));
+
+  isr.read();
+
+  BOOST_TEST(isr == 0x10);
+
+  device.close();
+}
+/**********************************************************************************************************************/
+/*if IAR is present: INTC writes 1<<n the according bit mask to IAR and not to ISR*/
+
+BOOST_AUTO_TEST_CASE(testIERwithIAR) { // TEST1
+  ChimeraTK::Device device;
+  device.open("(dummy:xdma/slot5?map=irq_test.mapp)");
+  BOOST_TEST(device.isOpened() == true);
+  auto accInterrupt = device.getVoidRegisterAccessor("!1:4", {ChimeraTK::AccessMode::wait_for_new_data});
+  auto dummyInterrupt = device.getVoidRegisterAccessor("DUMMY_INTERRUPT_1"); // for triggering the dummy interrupt
+
+  device.activateAsyncRead();
+
+  BOOST_TEST(readWithTimeout(accInterrupt)); // initial value after activation
+
+  auto ier = device.getScalarRegisterAccessor<uint32_t>("TEST1/IER");
+
+  ier.read();
+
+  BOOST_TEST(0x10 == ier);
+
+  auto isr = device.getScalarRegisterAccessor<uint32_t>("TEST1/ISR");
+  auto iar = device.getScalarRegisterAccessor<uint32_t>("TEST1/IAR");
+
+  isr = 0x110;
+  isr.write();
+
+  dummyInterrupt.write();
+
+  // wait until interrupt handler is done
+  BOOST_TEST(readWithTimeout(accInterrupt));
+
+  iar.read();
+
+  BOOST_TEST(
+      iar == 0x10); // we just hard coded this to 0x110, while in the map file its 0x0090000C. Why would it be 0x10?
+
+  // BOOST_TEST((accIAR >> 3) & 1) == true);
+
+  device.close();
+}
+/**********************************************************************************************************************/
+/* if ICR is present: INTC writes 1<<n the according bit mask to ICR and not to ISR */
+
+BOOST_AUTO_TEST_CASE(testIERwithICR) { // TEST2
```

Comment:
This test has 100 % redundancy with test IAR. Remove the code duplication. Cross-Check with the ISR case. The are mostly the same as well, especially if you consider the missing multi-accessor cases.

Reply:
done